test/contracts/LSP.spec.ts

```diff
@@ -2,28 +2,33 @@
  * @jest-environment node
  */
 
-import { ethers, getNamedAccounts } from "hardhat"
+import { config, ethers, getNamedAccounts } from "hardhat"
 import { waffleJest } from "@ethereum-waffle/jest"
 import { Address } from "hardhat-deploy/dist/types"
 import { BigNumber, Contract } from "ethers"
 import WETHAbi from "~/abis/WETH.json"
 import LSPABI from "~/abis/LSP.json"
 import LSPCreatorABI from "~/abis/LSPCreator.json"
 import ERC20ABI from "~/abis/ERC20.json"
+import OptimisticOracle from "~/abis/OptomisticOracle.json"
 import LinearLongShortPairABI from "~/abis/LinearLongShortPairFinancialProductLibrary.json"
 
 jest.setTimeout(40000)
 expect.extend(waffleJest)
 
 describe("LSP", function () {
-  const addresses: Record<string, Address> = {
+  const addresses = {
     WETH: "0xd0a1e359811322d97991e03f863a0c30c2cf029c",
     LSPCreator: "0x4C68829DBD07FEbB250B90f5624d4a5C30BBeC2c",
     LinearLongShortPairFinancialProductLibrary:
       "0x46b541E0fE2E817340A1A88740607329fF5ED279",
+    OptimisticOracle: "0xB1d3A89333BBC3F5e98A991d6d4C1910802986BC",
+    LSP: "",
+    longToken: "",
+    shortToken: "",
   }
-  const contracts: Record<string, Contract> = {}
-  let namedAccounts: Record<string, Address>
+  let contracts: Record<keyof typeof addresses, Contract>
```

Comment:
I can't run any tests because it complains that `contracts` is undefined.
I think you will have to initialize the variable here. (See old version)
![image](https://user-images.githubusercontent.com/15629702/123537590-8255df80-d730-11eb-9313-a626a45a7f80.png)

Reply:
Unfortunately after changing that I still get a bunch of errors due to contract objects being undefined. Were you able to run the tests on your machine ? (If so what is the command you used?)